### PR TITLE
toml-cli: init at 0.2.3

### PR DESCRIPTION
**Pull request by phlip9**
## Description of changes

Adds the [`toml-cli`](https://github.com/gnprice/toml-cli) tool at version 0.2.3

> a simple CLI for editing and querying TOML files.
>
> useful
> - in shell scripts, for consulting or editing a config file
> - and in instructions a human can follow for editing a config file, as a command to copy-paste and run.
>
> A key property is that when editing, we seek to preserve formatting and comments -- the only change to the file should be the one the user specifically asked for

### Example usage

```bash
$ nix shell github:phlip9/nixpkgs/toml-cli#toml-cli
$ toml --version
toml-cli 0.2.3

$ cat <<EOF > foo.toml
# This is a toml file
key = "value"
int = 17

# It has some comments
[foo]
x = "foo-x"
y.yy = "foo-yy"
EOF

$ toml get foo.toml .
{"key":"value","int":17,"foo":{"x":"foo-x","y":{"yy":"foo-yy"}}}

$ toml get -r foo.toml foo.y.yy
foo-yy

$ toml set foo.toml foo.x "cool new value"
# This is a toml file
key = "value"
int = 17

# It has some comments
[foo]
x = "cool new value"
y.yy = "foo-yy"
```

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wentasah** [comment]
Tested on `x86_64-linux`. LGTM.

**Review by wentasah — APPROVED** [review]
Approving based on the above comment.

File: pkgs/by-name/to/toml-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ lib, fetchCrate, rustPlatform, testers, toml-cli }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "toml-cli";
+  version = "0.2.3";
+
+  src = fetchCrate {
+    inherit pname version;
```
Comment (katexochen) [inline pkgs/by-name/to/toml-cli/package.nix:8]:
```suggestion
    inherit version;
    pname = "toml-cli";
```
Allow overriding pname.

**Review by katexochen — CHANGES_REQUESTED** [review]
just a nit, looks good otherwise.

**Comment by phlip9** [comment]
Applied suggestions & rebased. Should be good to merge!

**Review by katexochen — APPROVED** [review]

**Pull request merged**